Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -4663,5 +4743,18 @@ bool RenderBlockFlow::tryComputePreferredWidthsUsingModernPath(LayoutUnit& minLo
     return true;
 }
 
+LayoutUnit RenderBlockFlow::blockFormattingContextInFlowContentHeight() const
+{
+    ASSERT(createsBlockFormattingContext());
+    if (childrenInline())
+        return 0_lu;
+    // For block layout we should just be able to check the height of the last in flow box
+    auto lastChild = lastInFlowChildBox();
+    if (!lastChild)
+        return 0_lu;
+    // Child's margin box starting location + border box height + margin after size
+    return lastChild->y() + lastChild->height() + lastChild->marginAfter(&style());
```

Comment:
don't we have computed margin box geometry by the time we get here?

Reply:
how does this work with vertical writing-mode?

Reply:
Changed to check `isHorizontalWritingMode` first and then using `lastChild->marginBoxRect().height/width`